### PR TITLE
Week 15 – Labyrinth Game – Emma and Jin 

### PR DESCRIPTION
**Pull request by emmajosefina**
https://labyrinth-game-emma-jin.netlify.app/

### CODE REVIEW

File: code/src/components/Congrats.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  loop: true,
+  autoplay: true,
+  animationData,
+  renderSettings: {
+    preserveAspectRation: "xMidYMid slice",
+  },
+}
+
+const Congrats = () => (
+  <StyledCongrats>
```
Comment (idanaslund) [inline code/src/components/Congrats.js:16]: Really great with the animation at the end!

File: code/src/components/GamePlay.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+  return (
+    <GameArea>
+      {isLoading ? (
+        <Loading />
+      ) : (
+        <ContainerWrapper>
+          <Container>
+            <h3>{gameObject.description}</h3>
+            <StyledMsg>- where to go next?</StyledMsg>
```
Comment (idanaslund) [inline code/src/components/GamePlay.js:52]: Nice touch with your own text here!

File: code/src/components/Grid.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+import { useSelector } from "react-redux"
+
+const Grid = () => {
+  const coordinates = useSelector((state) => state.game.gameObject.coordinates)
+  const history = useSelector((state) => state.game.history)
+  const coordHist = history.map((item) => item.coordinates)
+  const x = Number(coordinates.split(",")[0])
+  const y = Number(coordinates.split(",")[1])
+
+  return (
```
Comment (idanaslund) [inline code/src/components/Grid.js:12]: What a way to think outside of the box! Really impressed by this, good job!

File: code/src/components/StartScreen.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+              type="text"
+              value={nameInput}
+              placeholder="Enter name here"
+              onChange={onSetNameInput}
+              onKeyDown={(event) => {
+                onEnter(event)
+              }}
+            />
+          </label>
+          <StyledButton unclickable={!nameInput} onClick={() => onStart()}>
```
Comment (idanaslund) [inline code/src/components/StartScreen.js:66]: `unclickable` was new to me! Have only used `disabled` before but will remember this one.

**Review by idanaslund — APPROVED** [review]
Great work this week! Your project is not just pretty with nice design and animations but it also has alot of interesting extras! The grid especially was really cool. 

One comment I have is that the grid could have been displayed to the side instead of in a column, when viewed on desktop, so that it doesn't disappear from the screen. 

Overall you have followed the Technigo guidelines for good code and achieved blue, red and even some of the black requirements! Feel proud of this project.

File: code/public/index.html [hunk 5]
```diff
@@ -1,19 +1,28 @@
...
+    <link
+      href="https://fonts.googleapis.com/css2?family=Cormorant:ital,wght@1,300&family=Roboto:wght@300;500&display=swap"
+      rel="stylesheet"
+    />
+
+      <!-- OG TAGS – SOCIAL MEDIA -->
+      <meta property="og:title" content="Labyrinth Game" />
+      <meta property="og:description" content="Made with Redux, APIs and React" />
+      <meta property="og:image" content="https://i.postimg.cc/G2HhrVBd/labyrinth-game.png" />
+      <meta property="og:url" content="https://labyrinth-game-emma-jin.netlify.app/" />
```
Comment (Neaa99) [inline code/public/index.html:19]: Woho OG tags, love them!

File: code/src/App.js [hunk 6]
```diff
@@ -1,9 +1,29 @@
...
-      Find me in src/app.js!
-    </div>
+    <Provider store={store}>
+      <BrowserRouter>
+        <Routes>
+          <Route path="/" element={<StartScreen />} />
+          <Route path="/Gameplay" element={<GamePlay />} />
+        </Routes>
+      </BrowserRouter>
+    </Provider>
```
Comment (Neaa99) [inline code/src/App.js:27]: Nice and clean!

File: code/src/components/Congrats.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import Lottie from "react-lottie"
+import animationData from "../lotties/congrats"
+import { StyledCongrats } from "styles"
+
+const defaultOptions = {
+  loop: true,
+  autoplay: true,
+  animationData,
+  renderSettings: {
+    preserveAspectRation: "xMidYMid slice",
```
Comment (Neaa99) [inline code/src/components/Congrats.js:11]: Nice with the lottie! Havn't tried that out yet, i'm to scared haha. But your animation looks amazing!

File: code/src/components/GamePlay.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    dispatch(reset())
+    navigate("/")
+  };
+
+  const onGoBack = () => {
+    if (!history.length) {
+      return
+    }
+    dispatch(setPreviousGameObject())
+  };
```
Comment (Neaa99) [inline code/src/components/GamePlay.js:42]: Really great with a Go Back button! I'm impressed

**Review by Neaa99 — COMMENTED** [review]
Jin and Emma, WOW! Your page looks och works amazing. All of your code is clean and you added a lot of extra work with the Go Back, Grid and Lotties! Learned a lot from reading your code, great job! If I'm not wrong you have reached the Red level, I can't tell if you have used to design libary or not. 

Amazing work!

File: code/src/components/Congrats.js (hunk 7 above)
Comment (JenFi72) [inline code/src/components/Congrats.js:11]: Good job with the animation at the end! :)

File: code/src/components/GamePlay.js [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+import {
+  GameArea,
+  StyledButton,
+  StyledButtonB,
+  Container,
+  ContainerWrapper,
+  Wrapper,
+  StyledMsg,
+} from "styles"
```
Comment (JenFi72) [inline code/src/components/GamePlay.js:15]: Good way of keeping the code very clean and easy to follow :)

File: code/src/components/GamePlay.js [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+                      {action.direction +
+                        (action.direction === "North"
+                          ? " ⬆"
+                          : action.direction === "South"
+                          ? " ⬇"
+                          : action.direction === "West"
+                          ? " ⬅"
+                          : action.direction === "East"
+                          ? " ➡"
+                          : "")}
```
Comment (JenFi72) [inline code/src/components/GamePlay.js:76]: Nice to add arrows to buttons

File: code/src/components/GamePlay.js [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+                    </span>
+                  </StyledButton>
+                </Container>
+              ))}
+              <Wrapper>
+                <StyledButtonB onClick={onRestartClick}>RESTART</StyledButtonB>
+                <StyledButtonB unclickable={!history.length} onClick={onGoBack}>
+                  GO BACK
```
Comment (JenFi72) [inline code/src/components/GamePlay.js:84]: "Simple" code, but still adds great features to the game :)

File: code/src/components/Grid.js [hunk 12]
```diff
@@ -0,0 +1,201 @@
+import React from "react"
+import { StyledGrid, StyledGridItem } from "styles"
+import { useSelector } from "react-redux"
+
+const Grid = () => {
+  const coordinates = useSelector((state) => state.game.gameObject.coordinates)
+  const history = useSelector((state) => state.game.history)
+  const coordHist = history.map((item) => item.coordinates)
+  const x = Number(coordinates.split(",")[0])
```
Comment (JenFi72) [inline code/src/components/Grid.js:9]: Impressed you managed to this by yourself!!

File: code/src/styles.js [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+`
+
+export const ContainerWrapper = styled.div`
+  padding-top: 5%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+`
+
```
Comment (JenFi72) [inline code/src/styles.js:27]: Very good structure :)

**Review by JenFi72 — COMMENTED** [review]
Such an impressive work!!! 

The game looks great as you've added nice styling, animations and your own labyrinth following the actions the player choses to take. Great! Code is also super structured and clean. 

If I would say some feedback, it would be to change some styling dependent on device. When playing from my PC, the labyrinth isn't always showing. But overall VERY good job!!!

**Pull request closed without merging**